### DISCUSSION TITLE
How to upload a file from a blade component?

### DISCUSSION THREAD

**Discussion by Muratcol** [orchidsoftware/platform#2013] [Q&A]
II created a custom blade template to send email. It works very well, but I need to upload html files and I couldn't find a way to do it.

This is the code part I try to upload file:

```HTML
...
                <div class="uploadTemplateBlock">
                    
                    <form action="{{ route('template.upload') }}" method="POST" enctype="multipart/form-data">
                        @csrf
                        <input class="uploadButtons" type="file" id="mustacheTemplate" name="mustacheTemplate" onchange="this.form.submit()">
                    </form>
                </div>
...
```

And this is the `template.upload` route

```php
web.php
Route::post('/templateUpload', 'App\Http\Controllers\TemplateUpload@upload')->name('template.upload');
```

When I select a file with to upload I'm having this error:

```
TypeError
Argument 1 passed to Orchid\Screen\Screen::callMethod() must be of the type string, null given, called in
C:\Users\murat\OneDrive\Desktop\status-dashboard\laravel\vendor\orchid\platform\src\Screen\Screen.php
on line 175
```

Any suggestion?

**Comment by Muratcol** [comment]
Do you have any suggestions on this senpai  @tabuna ? 😄

**Reply by tabuna** [reply]
Hey @Muratcol, The screen [already contains the base `<form>` tag](https://github.com/orchidsoftware/platform/blob/099c00336afabce50c26c1bb3fd8d22be1eba53f/resources/views/layouts/base.blade.php#L40) so it didn't work for you. I don't have a good solution for this situation because a form cannot be nested in another form
